src/darwin/Framework/CHIP/MTRDeviceControllerFactory.mm

```diff
@@ -891,13 +893,12 @@ - (void)resetOperationalAdvertising
     if (_controllers.count != 0) {
         // We have a running controller.  That means we likely need to reset
         // operational advertising for that controller.
-        dispatch_async(_chipWorkQueue, ^{
-            // StartServer() is the only API we have for resetting DNS-SD
-            // advertising.  It sure would be nice if there were a "restart"
-            // that was a no-op if the DNS-SD server was not already
-            // running.
-            app::DnssdServer::Instance().StartServer();
-        });
+
+        // StartServer() is the only API we have for resetting DNS-SD
+        // advertising.  It sure would be nice if there were a "restart"
+        // that was a no-op if the DNS-SD server was not already
+        // running.
+        app::DnssdServer::Instance().StartServer();
```

Comment:
Though I would be a bit happier if we did a "we have a running controller" check under the lock, then released the lock before we call into arbitrary code via StartServer() here.  We know the "have a running controller" state can't change while we are in this function, even if we release the controllers lock, because nothing else can run on the Matter queue while we are here, with the dispatch now gone.